tokio/src/runtime/basic_scheduler.rs

```diff
@@ -152,7 +152,7 @@ where
                     };
 
                     match next {
-                        Some(task) => task.run(),
+                        Some(task) => crate::coop::budget(|| task.run()),
```

Comment:
The call to `budget` had to be removed from the task abstraction as we don't **always** want to reset the budget. This seemed simpler than adding APIs to get the remainder of the budget and pass a budget argument to the task.

Reply:
@carllerche With the call to `budget` in the task abstraction, that was resetting the worker's budget each time any task actually ran (especially in the ping ping situation)?

Reply:
Yes, but moving it out into the scheduler is functionally exactly the same. The downside is that the call must be repeated once per scheduler. This is necessary as the required behavior is no longer identical across schedulers.

Reply:
Yep that makes sense 👍